src/test/java/org/folio/rest/impl/SynchronizationAPITests.java

```diff
@@ -160,6 +169,37 @@ public void declaredLostEventShouldBeCreatedAfterSynchronization() {
     checkSyncJobUpdatedByLoanEvent(syncJobId);
   }
 
+  @Test
+  public void agedToLostEventShouldBeDeletedBeforeSynchronizationJobByUser() {
+    sendEvent(buildItemAgedToLostEvent(USER_ID, randomId()), SC_NO_CONTENT);
+    assertThat(waitFor(itemAgedToLostEventRepository.getByUserId(USER_ID)).size(), is(1));
+    String syncJobId = createOpenSynchronizationJobByUser();
+
+    runSynchronization();
+
+    Awaitility.await()
+      .atMost(5, SECONDS)
+      .until(() -> waitFor(itemAgedToLostEventRepository.getByUserId(USER_ID)).size(), is(0));
+
```

Comment:
If I remove outer await, 3 sec on waitFor() is not enough for whole synchronization method, and returns failing result, here returns size == 1

Reply:
Try increasing the timeout in `waitFor`.

Reply:
@alexanderkurash we do need this nested `waitFor` here. Awaitility polls the `Supplier` passed into `until()` method until the condition represented by the matcher is satisfied (or util the waiting time runs out). But since we supply an asynchronous operation which returns a `Future`, we need to wait until it's complete before checking the result. On a `CompletableFuture` we could just call the blocking `get()`, but Vertx `Future` will return `null` immediately, so we have to block execution manually. One option is to convert the `Future` to `CompletableFuture` (there are built-in methods for this) and call `get()`, but using `waitFor` is more consise. So I suggest leaving it as is until we come up with a better solution.